### PR TITLE
Fix outdated remote resolution documentation

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

This commit fixes a few issues with our docs on remote resolution:
- Update feature stability from alpha to beta
- Add a section on specifying remote tasks in pipelines
- Fix outdated configuration/install docs to indicate resolvers are enabled by default
- Define resolution and link to the TEP

Partially addresses #6045

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/pipelines.md [hunk 1]
```diff
@@ -316,6 +317,28 @@ tasks:
...
+- name: "go-build"
+  taskRef:
+    resolver: git
+    params:
+    - name: url
+      value: https://github.com/tektoncd/catalog.git
+    - name: revision
+      value: $(params.gitRevision) # can use params declared at the pipeline level
+    - name: pathInRepo
+      value: /task/golang-build/0.3/golang-build.yaml
```
Comment (lbernick) [inline docs/pipelines.md:339]:
```suggestion
      value: task/golang-build/0.3/golang-build.yaml
```

**Review by lbernick — APPROVED** [review]

File: docs/pipelines.md [hunk 2]
```diff
@@ -316,6 +317,28 @@ tasks:
...
+```yaml
+tasks:
+- name: "go-build"
+  taskRef:
+    resolver: git
+    params:
+    - name: url
+      value: https://github.com/tektoncd/catalog.git
+    - name: revision
+      value: $(params.gitRevision) # can use params declared at the pipeline level
```
Comment (XinruZhang) [inline docs/pipelines.md:337]: Nit: Wondeing if using "main" would be easier for users to use this example code for reference. 🤔

File: docs/resolution.md [hunk 3]
```diff
@@ -1,23 +1,18 @@
...
-1. [The `bundles` resolver](./bundle-resolver.md), enabled by setting the `enable-bundles-resolver`
-   feature flag to `true`.
-1. [The `git` resolver](./git-resolver.md), enabled by setting the `enable-git-resolver`
-   feature flag to `true`.
-1. [The `hub` resolver](./hub-resolver.md), enabled by setting the `enable-hub-resolver`
-   feature flag to `true`.
-1. [The `cluster` resolver](./cluster-resolver.md), enabled by setting the `enable-cluster-resolver`
-   feature flag to `true`.
+These resolvers are enabled by setting the appropriate feature flag in the `resolvers-feature-flags`
+ConfigMap in the `tekton-pipelines-resolvers` namespace. See the [section in install.md](install.md#configuring-built-in-remote-task-and-pipeline-resolution) for details/
```
Comment (XinruZhang) [inline docs/resolution.md:15]: an extra `/` at the end :)

**Review by XinruZhang — COMMENTED** [review]
Thanks @dibyom! just some minor comments :)

File: docs/pipelines.md (hunk 2 above)
Comment (dibyom) [reply docs/pipelines.md:337]: I added this deliberately to show one could use params for the value here. I think pipeline authors should be familiar with params so hopefully it isn't too confusing

Comment (dibyom) [reply docs/pipelines.md:337]: Added a more detailed comment

**Pull request merged**